lib/ohai/plugins/rpm.rb

```diff
@@ -91,17 +91,25 @@
       value = ""
       lines[macros_start_idx + 1..macros_end_idx - 1].each do |line|
         if line.start_with?("-")
+          # new macros are always prefixed by a dash
           if name
+            # if we had parsed a macro before, store it
             rpm[:macros][name] = value
             name = nil
             value = ""
-          else
-            _prefix, name, value = line.split(" ", 3)
           end
+
+          # parse the new macro definition
+          _prefix, name, value = line.split(" ", 3)
         else
+          # continuations have no prefix and just append to the previous one
           value += "\n#{line}"
         end
       end
+      if name
+        # ensure the last macro definition is also stored
+        rpm[:macros][name] = value
+      end
```

Comment:
The comment where it is and the loop being big enough, I'd say this is a bit unclear. Perhaps change it this way.

```suggestion

      # Once we're done parsing all lines, we might have a parsed definition
      # we haven't stored - if so, store it.
      if name
        rpm[:macros][name] = value
      end
```

Reply:
Comment updated, thanks!